### ISSUE TITLE
Alert is not recognized on  Xcode 11 beat4 /iOS 13 

### ISSUE THREAD

**Issue by bearxc** [appium/appium#12957]
iOS 13  simulator on Xcode 11 beta 4 
appium : latest master branch
java client: 7.1.0

when I try to get UI dump through

Context.driver.getPageSource();

I can not find any XCUIElementTypeAlert related info , text/buttons inside alert is missing also .

**Comment by KazuCocoa** [comment]
Could you attach the appium log following issue template?
`source` result also helps us.

**Comment by bearxc** [comment]
Following is   appium log ,   there is error msg   ,



 WebDriverAgentRunner-Runner[23469:276301] Internal error: Error Domain=XCTDaemonErrorDomain Code=39 "Legacy accessibility snapshots are no longer supported" UserInfo={NSLocalizedDescription=Legacy accessibility snapshots are no longer supported}

 I opened  another one as 
https://github.com/appium/appium/issues/12960



2019-07-23 22:38:40:704 - [HTTP] --> GET /wd/hub/session/8cba7c6f-c019-4ca4-9426-4c62cd8d2cae/source
2019-07-23 22:38:40:705 - [HTTP] {}
2019-07-23 22:38:40:705 - [debug] [W3C (8cba7c6f)] Calling AppiumDriver.getPageSource() with args: ["8cba7c6f-c019-4ca4-9426-4c62cd8d2cae"]
2019-07-23 22:38:40:705 - [debug] [XCUITest] Executing command 'getPageSource'
2019-07-23 22:38:40:707 - [debug] [WD Proxy] Matched '/source' to command name 'getPageSource'
2019-07-23 22:38:40:707 - [debug] [WD Proxy] Proxying [GET /source] to [GET http://localhost:8100/session/6143B327-6099-43E5-8A08-1C1888E37DE5/source] with no body
2019-07-23 22:38:40:773 - [Xcode] 2019-07-23 22:38:40.773070-0400 WebDriverAgentRunner-Runner[23469:276301] Internal error: Error Domain=XCTDaemonErrorDomain Code=39 "Legacy accessibility snapshots are no longer supported" UserInfo={NSLocalizedDescription=Legacy accessibility snapshots are no longer supported}
2019-07-23 22:38:40:773 - [Xcode]
2019-07-23 22:38:40:774 - [Xcode]     t =    71.34s Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:774 - [Xcode]
2019-07-23 22:38:40:774 - [Xcode]     t =    71.34s     Returning cached snapshot
2019-07-23 22:38:40:775 - [Xcode]
2019-07-23 22:38:40:776 - [Xcode]     t =    71.34s Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:776 - [Xcode]
2019-07-23 22:38:40:776 - [Xcode]     t =    71.34s     Returning cached snapshot
2019-07-23 22:38:40:776 - [Xcode]
2019-07-23 22:38:40:781 - [Xcode]     t =    71.35s Get all elements bound by accessibility element for: Elements matching predicate 'wdUID IN {"05000000-0000-0000-AF5B-000000000000", "1C000000-0000-0000-AF5B-000000000000"} AND (1 == 1 OR identifier == 0 OR frame == 0 OR value == 0 OR title == 0 OR label == 0 OR elementType == 0 OR enabled == 0 OR placeholderValue == 0)'
2019-07-23 22:38:40:781 - [Xcode]
2019-07-23 22:38:40:782 - [Xcode]     t =    71.35s     Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:783 - [Xcode]
2019-07-23 22:38:40:816 - [Xcode]     t =    71.38s     Find: Descendants matching type Window
2019-07-23 22:38:40:817 - [Xcode]
2019-07-23 22:38:40:818 - [Xcode]     t =    71.38s     Find: Elements matching predicate 'wdUID IN {"05000000-0000-0000-AF5B-000000000000", "1C000000-0000-0000-AF5B-000000000000"} AND (1 == 1 OR identifier == 0 OR frame == 0 OR value == 0 OR title == 0 OR label == 0 OR elementType == 0 OR enabled == 0 OR placeholderValue == 0)'
2019-07-23 22:38:40:818 - [Xcode]
2019-07-23 22:38:40:819 - [Xcode]     t =    71.39s Checking existence of `Window`
2019-07-23 22:38:40:819 - [Xcode]
2019-07-23 22:38:40:820 - [Xcode]     t =    71.39s     Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:820 - [Xcode]
2019-07-23 22:38:40:849 - [Xcode]     t =    71.42s     Find: Descendants matching type Window
2019-07-23 22:38:40:849 - [Xcode]
2019-07-23 22:38:40:850 - [Xcode]     t =    71.42s     Find: Elements matching predicate 'wdUID IN {"05000000-0000-0000-AF5B-000000000000", "1C000000-0000-0000-AF5B-000000000000"} AND (1 == 1 OR identifier == 0 OR frame == 0 OR value == 0 OR title == 0 OR label == 0 OR elementType == 0 OR enabled == 0 OR placeholderValue == 0)'
2019-07-23 22:38:40:850 - [Xcode]
2019-07-23 22:38:40:851 - [Xcode]     t =    71.42s     Find: Identity Binding
2019-07-23 22:38:40:851 - [Xcode]
2019-07-23 22:38:40:852 - [Xcode]     t =    71.42s Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:853 - [Xcode]
2019-07-23 22:38:40:853 - [Xcode]     t =    71.42s     Returning cached snapshot
2019-07-23 22:38:40:853 - [Xcode]
2019-07-23 22:38:40:854 - [Xcode]     t =    71.42s Find: Descendants matching type Window
2019-07-23 22:38:40:854 - [Xcode]
2019-07-23 22:38:40:855 - [Xcode]     t =    71.42s Find: Elements matching predicate 'wdUID IN {"05000000-0000-0000-AF5B-000000000000", "1C000000-0000-0000-AF5B-000000000000"} AND (1 == 1 OR identifier == 0 OR frame == 0 OR value == 0 OR title == 0 OR label == 0 OR elementType == 0 OR enabled == 0 OR placeholderValue == 0)'
2019-07-23 22:38:40:855 - [Xcode]
2019-07-23 22:38:40:856 - [Xcode]     t =    71.42s Find: Identity Binding
2019-07-23 22:38:40:856 - [Xcode]
2019-07-23 22:38:40:857 - [Xcode]     t =    71.42s Checking existence of `Window`
2019-07-23 22:38:40:857 - [Xcode]
2019-07-23 22:38:40:858 - [Xcode]     t =    71.43s     Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:858 - [Xcode]
2019-07-23 22:38:40:884 - [Xcode]     t =    71.45s     Find: Descendants matching type Window
2019-07-23 22:38:40:884 - [Xcode]
2019-07-23 22:38:40:885 - [Xcode]     t =    71.45s     Find: Elements matching predicate 'wdUID IN {"05000000-0000-0000-AF5B-000000000000", "1C000000-0000-0000-AF5B-000000000000"} AND (1 == 1 OR identifier == 0 OR frame == 0 OR value == 0 OR title == 0 OR label == 0 OR elementType == 0 OR enabled == 0 OR placeholderValue == 0)'
2019-07-23 22:38:40:885 - [Xcode]
2019-07-23 22:38:40:886 - [Xcode]     t =    71.45s     Find: Identity Binding
2019-07-23 22:38:40:886 - [Xcode]
2019-07-23 22:38:40:888 - [Xcode]     t =    71.46s Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:888 - [Xcode]
2019-07-23 22:38:40:888 - [Xcode]     t =    71.46s     Returning cached snapshot
2019-07-23 22:38:40:888 - [Xcode]
2019-07-23 22:38:40:889 - [Xcode]     t =    71.46s Find: Descendants matching type Window
2019-07-23 22:38:40:890 - [Xcode]
2019-07-23 22:38:40:891 - [Xcode]     t =    71.46s Find: Elements matching predicate 'wdUID IN {"05000000-0000-0000-AF5B-000000000000", "1C000000-0000-0000-AF5B-000000000000"} AND (1 == 1 OR identifier == 0 OR frame == 0 OR value == 0 OR title == 0 OR label == 0 OR elementType == 0 OR enabled == 0 OR placeholderValue == 0)'
2019-07-23 22:38:40:891 - [Xcode]
2019-07-23 22:38:40:892 - [Xcode]     t =    71.46s Find: Identity Binding
2019-07-23 22:38:40:892 - [Xcode]
2019-07-23 22:38:40:893 - [Xcode]     t =    71.46s Find the Window
2019-07-23 22:38:40:893 - [Xcode]
2019-07-23 22:38:40:895 - [Xcode]     t =    71.46s     Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:895 - [Xcode]
2019-07-23 22:38:40:920 - [Xcode]     t =    71.49s     Find: Descendants matching type Window
2019-07-23 22:38:40:920 - [Xcode]
2019-07-23 22:38:40:921 - [Xcode]     t =    71.49s     Find: Elements matching predicate 'wdUID IN {"05000000-0000-0000-AF5B-000000000000", "1C000000-0000-0000-AF5B-000000000000"} AND (1 == 1 OR identifier == 0 OR frame == 0 OR value == 0 OR title == 0 OR label == 0 OR elementType == 0 OR enabled == 0 OR placeholderValue == 0)'
2019-07-23 22:38:40:921 - [Xcode]
2019-07-23 22:38:40:922 - [Xcode]     t =    71.49s     Find: Identity Binding
2019-07-23 22:38:40:922 - [Xcode]
2019-07-23 22:38:40:924 - [Xcode] 2019-07-23 22:38:40.924126-0400 WebDriverAgentRunner-Runner[23469:276301] Cannot take the snapshot of Window at {{0.0, 0.0}, {375.0, 667.0}} after 15 seconds
2019-07-23 22:38:40:924 - [Xcode]
2019-07-23 22:38:40:924 - [Xcode] 2019-07-23 22:38:40.924265-0400 WebDriverAgentRunner-Runner[23469:276301] Internal error: Error Domain=XCTDaemonErrorDomain Code=39 "Legacy accessibility snapshots are no longer supported" UserInfo={NSLocalizedDescription=Legacy accessibility snapshots are no longer supported}
2019-07-23 22:38:40:925 - [Xcode]
2019-07-23 22:38:40:925 - [Xcode]     t =    71.49s Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:925 - [Xcode]
2019-07-23 22:38:40:925 - [Xcode]     t =    71.49s     Returning cached snapshot
2019-07-23 22:38:40:926 - [Xcode]
2019-07-23 22:38:40:926 - [Xcode]     t =    71.49s Find: Descendants matching type Window
2019-07-23 22:38:40:927 - [Xcode]
2019-07-23 22:38:40:927 - [Xcode]     t =    71.49s Find: Elements matching predicate 'wdUID IN {"05000000-0000-0000-AF5B-000000000000", "1C000000-0000-0000-AF5B-000000000000"} AND (1 == 1 OR identifier == 0 OR frame == 0 OR value == 0 OR title == 0 OR label == 0 OR elementType == 0 OR enabled == 0 OR placeholderValue == 0)'
2019-07-23 22:38:40:928 - [Xcode]
2019-07-23 22:38:40:928 - [Xcode]     t =    71.50s Find: Identity Binding
2019-07-23 22:38:40:929 - [Xcode]
2019-07-23 22:38:40:930 - [Xcode]     t =    71.50s Find the Window
2019-07-23 22:38:40:930 - [Xcode]
2019-07-23 22:38:40:931 - [Xcode]     t =    71.50s     Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:931 - [Xcode]
2019-07-23 22:38:40:958 - [Xcode]     t =    71.52s     Find: Descendants matching type Window
2019-07-23 22:38:40:958 - [Xcode]
2019-07-23 22:38:40:959 - [Xcode]     t =    71.53s     Find: Elements matching predicate 'wdUID IN {"05000000-0000-0000-AF5B-000000000000", "1C000000-0000-0000-AF5B-000000000000"} AND (1 == 1 OR identifier == 0 OR frame == 0 OR value == 0 OR title == 0 OR label == 0 OR elementType == 0 OR enabled == 0 OR placeholderValue == 0)'
2019-07-23 22:38:40:959 - [Xcode]
2019-07-23 22:38:40:960 - [Xcode]     t =    71.53s     Find: Identity Binding
2019-07-23 22:38:40:960 - [Xcode]
2019-07-23 22:38:40:963 - [Xcode] 2019-07-23 22:38:40.962691-0400 WebDriverAgentRunner-Runner[23469:276301] Cannot take the snapshot of Window at {{0.0, 0.0}, {375.0, 667.0}} after 15 seconds
2019-07-23 22:38:40:963 - [Xcode]
2019-07-23 22:38:40:963 - [Xcode] 2019-07-23 22:38:40.962898-0400 WebDriverAgentRunner-Runner[23469:276301] Internal error: Error Domain=XCTDaemonErrorDomain Code=39 "Legacy accessibility snapshots are no longer supported" UserInfo={NSLocalizedDescription=Legacy accessibility snapshots are no longer supported}
2019-07-23 22:38:40:963 - [Xcode]
2019-07-23 22:38:40:964 - [Xcode]     t =    71.53s Requesting snapshot of accessibility hierarchy for app with pid 23471
2019-07-23 22:38:40:964 - [Xcode]
2019-07-23 22:38:40:964 - [Xcode]     t =    71.53s     Returning cached snapshot
2019-07-23 22:38:40:964 - [Xcode]
2019-07-23 22:38:40:966 - [Xcode]     t =    71.53s Find: Descendants matching type Window
2019-07-23 22:38:40:966 - [Xcode]
2019-07-23 22:38:40:967 - [Xcode]     t =    71.53s Find: Elements matching predicate 'wdUID IN {"05000000-0000-0000-AF5B-000000000000", "1C000000-0000-0000-AF5B-000000000000"} AND (1 == 1 OR identifier == 0 OR frame == 0 OR value == 0 OR title == 0 OR label == 0 OR elementType == 0 OR enabled == 0 OR placeholderValue == 0)'
2019-07-23 22:38:40:967 - [Xcode]
2019-07-23 22:38:40:968 - [Xcode]     t =    71.54s Find: Identity Binding

**Comment by KazuCocoa** [comment]
> I can not find any XCUIElementTypeAlert related info , text/buttons inside alert is missing also .

Do you have any good example?
Below is an alert on Map app in iOS 13, iPhone 8 simulator with Xcode 11 beta 4.
I've published webdriveragent 0.5.0 in a couple of hours ago, but below could get with 0.4.0.


```
<?xml version="1.0" encoding="UTF-8"?>
<AppiumAUT>
  <XCUIElementTypeApplication type="XCUIElementTypeApplication" enabled="true" visible="true" x="0" y="0" width="375" height="667" name="Maps" label="Maps">
    <XCUIElementTypeWindow type="XCUIElementTypeWindow" enabled="true" visible="true" x="0" y="0" width="375" height="667">
      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="0" y="0" width="375" height="667">
        <XCUIElementTypeAlert type="XCUIElementTypeAlert" enabled="true" visible="true" x="52" y="183" width="271" height="322" name="Allow “Maps” to access your location while you are using the app?" label="Allow “Maps” to access your location while you are using the app?">
          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="183" width="271" height="322">
            <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="183" width="271" height="322">
              <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="52" y="183" width="271" height="322">
                <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="52" y="183" width="271" height="322">
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="52" y="183" width="271" height="322">
                  </XCUIElementTypeOther>
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="52" y="183" width="271" height="322">
                  </XCUIElementTypeOther>
                </XCUIElementTypeOther>
              </XCUIElementTypeOther>
              <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="183" width="271" height="322">
                <XCUIElementTypeScrollView type="XCUIElementTypeScrollView" enabled="true" visible="true" x="52" y="183" width="271" height="188">
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="183" width="271" height="188">
                    <XCUIElementTypeStaticText type="XCUIElementTypeStaticText" enabled="true" visible="true" x="68" y="203" width="239" height="65" name="Allow “Maps” to access your location while you are using the app?" label="Allow “Maps” to access your location while you are using the app?" value="Allow “Maps” to access your location while you are using the app?">
                    </XCUIElementTypeStaticText>
                    <XCUIElementTypeStaticText type="XCUIElementTypeStaticText" enabled="true" visible="true" x="68" y="270" width="239" height="81" name="App explanation for While Use App: “Your current location will be displayed on the map and used for directions, nearby search results, and estimated travel times.”" label="App explanation for While Use App: “Your current location will be displayed on the map and used for directions, nearby search results, and estimated travel times.”" value="App explanation for While Use App: “Your current location will be displayed on the map and used for directions, nearby search results, and estimated travel times.”">
                    </XCUIElementTypeStaticText>
                  </XCUIElementTypeOther>
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="316" y="183" width="4" height="188" name="Vertical scroll bar, showing 100% of the content" label="Vertical scroll bar, showing 100% of the content" value="0%">
                  </XCUIElementTypeOther>
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="52" y="365" width="271" height="3" name="Horizontal scroll bar, showing 100% of the content" label="Horizontal scroll bar, showing 100% of the content" value="0%">
                  </XCUIElementTypeOther>
                </XCUIElementTypeScrollView>
                <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="371" width="271" height="1">
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="371" width="271" height="1">
                    <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="371" width="271" height="1">
                    </XCUIElementTypeOther>
                  </XCUIElementTypeOther>
                </XCUIElementTypeOther>
                <XCUIElementTypeScrollView type="XCUIElementTypeScrollView" enabled="true" visible="true" x="52" y="371" width="271" height="134">
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="371" width="271" height="134">
                    <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="371" width="271" height="134">
                      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="371" width="271" height="45">
                        <XCUIElementTypeButton type="XCUIElementTypeButton" enabled="true" visible="true" x="52" y="371" width="271" height="45" name="Allow While Using App" label="Allow While Using App">
                        </XCUIElementTypeButton>
                      </XCUIElementTypeOther>
                      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="415" width="271" height="1">
                        <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="415" width="271" height="1">
                          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="415" width="271" height="1">
                          </XCUIElementTypeOther>
                        </XCUIElementTypeOther>
                      </XCUIElementTypeOther>
                      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="416" width="271" height="44">
                        <XCUIElementTypeButton type="XCUIElementTypeButton" enabled="true" visible="true" x="52" y="416" width="271" height="44" name="Allow Once" label="Allow Once">
                        </XCUIElementTypeButton>
                      </XCUIElementTypeOther>
                      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="460" width="271" height="1">
                        <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="460" width="271" height="1">
                          <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="460" width="271" height="1">
                          </XCUIElementTypeOther>
                        </XCUIElementTypeOther>
                      </XCUIElementTypeOther>
                      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="true" x="52" y="460" width="271" height="45">
                        <XCUIElementTypeButton type="XCUIElementTypeButton" enabled="true" visible="true" x="52" y="460" width="271" height="45" name="Don’t Allow" label="Don’t Allow">
                        </XCUIElementTypeButton>
                      </XCUIElementTypeOther>
                    </XCUIElementTypeOther>
                  </XCUIElementTypeOther>
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="316" y="371" width="4" height="134" name="Vertical scroll bar, showing 100% of the content" label="Vertical scroll bar, showing 100% of the content" value="0%">
                  </XCUIElementTypeOther>
                  <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="52" y="498" width="271" height="4" name="Horizontal scroll bar, showing 100% of the content" label="Horizontal scroll bar, showing 100% of the content" value="0%">
                  </XCUIElementTypeOther>
                </XCUIElementTypeScrollView>
              </XCUIElementTypeOther>
            </XCUIElementTypeOther>
          </XCUIElementTypeOther>
        </XCUIElementTypeAlert>
      </XCUIElementTypeOther>
    </XCUIElementTypeWindow>
    <XCUIElementTypeWindow type="XCUIElementTypeWindow" enabled="true" visible="false" x="0" y="0" width="375" height="667">
      <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="0" width="375" height="667">
        <XCUIElementTypeOther type="XCUIElementTypeOther" enabled="true" visible="false" x="0" y="0" width="375" height="667">
        </XCUIElementTypeOther>
      </XCUIElementTypeOther>
    </XCUIElementTypeWindow>
  </XCUIElementTypeApplication>
</AppiumAUT>
```

<img src="https://user-images.githubusercontent.com/5511591/61781550-8344af00-ae3f-11e9-89ff-b47fa4e446de.png" width=300>

**Comment by bearxc** [comment]
I think that is related with our own app issue .
